components/binding-usage-controller/internal/controller/supervisor_aggregator_test.go

```diff
@@ -9,111 +9,37 @@ import (
-		mocks[k] = m
-	}
+	fnSupervisor := &automock.KubernetesResourceSupervisor{}
+	deploySupervisor := &automock.KubernetesResourceSupervisor{}
 
-	// register all mock
 	aggregator := controller.NewResourceSupervisorAggregator()
-	for kind, mock := range mocks {
-		aggregator.Register(kind, mock)
-	}
-
-	for _, k := range registeredKinds {
-		// when
-		concrete, err := aggregator.Get(k)
-
-		// then
-		require.NoError(t, err)
-		assert.True(t, concrete.HasSynced())
-		mocks[k].AssertExpectations(t)
-	}
-}
-
-func TestResourceSupervisorAggregatorHasSynced(t *testing.T) {
-	tests := map[string]struct {
-		givenMocks map[controller.Kind]*automock.KubernetesResourceSupervisor
-		expSynced  bool
-	}{
-		"all synced": {
-			givenMocks: func() map[controller.Kind]*automock.KubernetesResourceSupervisor {
-				registeredKinds := []controller.Kind{
-					controller.KindKubelessFunction,
-					controller.KindDeployment,
-				}
-				mocks := make(map[controller.Kind]*automock.KubernetesResourceSupervisor)
-				for _, k := range registeredKinds {
-					m := &automock.KubernetesResourceSupervisor{}
-					m.ExpectOnHasSynced(true).Once()
-					mocks[k] = m
-				}
-				return mocks
-			}(),
-			expSynced: true,
-		},
-		"one not synced": {
-			givenMocks: func() map[controller.Kind]*automock.KubernetesResourceSupervisor {
-				mocks := make(map[controller.Kind]*automock.KubernetesResourceSupervisor)
-				d := &automock.KubernetesResourceSupervisor{}
-				d.ExpectOnHasSynced(false).Once()
-				mocks[controller.KindDeployment] = d
-
-				fn := &automock.KubernetesResourceSupervisor{}
-				fn.ExpectOnHasSynced(true).Once()
-				mocks[controller.KindKubelessFunction] = fn
-				return mocks
-			}(),
-			expSynced: false,
-		},
-		"all not synced": {
-			givenMocks: func() map[controller.Kind]*automock.KubernetesResourceSupervisor {
-				registeredKinds := []controller.Kind{
-					controller.KindKubelessFunction,
-					controller.KindDeployment,
-				}
-				mocks := make(map[controller.Kind]*automock.KubernetesResourceSupervisor)
-				for _, k := range registeredKinds {
-					m := &automock.KubernetesResourceSupervisor{}
-					m.ExpectOnHasSynced(false).Once()
-					mocks[k] = m
-				}
-				return mocks
-			}(),
-			expSynced: false,
-		},
-	}
-	for tn, tc := range tests {
-		t.Run(tn, func(t *testing.T) {
-			// given
-			// register all mock
-			aggregator := controller.NewResourceSupervisorAggregator()
-			for kind, mock := range tc.givenMocks {
-				aggregator.Register(kind, mock)
-			}
+	aggregator.Register(kindFunction, fnSupervisor)
+	aggregator.Register(kindDeployment, deploySupervisor)
 
-			assert.Equal(t, tc.expSynced, aggregator.HasSynced())
-		})
-	}
+	// when
+	gotFn, err := aggregator.Get(kindFunction)
+	require.NoError(t, err)
+	gotDeploy, err := aggregator.Get(kindDeployment)
+	require.NoError(t, err)
 
+	// then
+	assert.True(t, fnSupervisor == gotFn)
+	assert.True(t, deploySupervisor == gotDeploy)
```

Comment:
Isn't it better to use `assert.Equal` here?

Reply:
it is because I want to assert if it is about the same objects, I don't want to compare fields